wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController+Scrolling.swift

```diff
@@ -52,11 +52,22 @@ extension ConversationContentViewController {
         updateTableViewHeaderView()
     }
 
-    func scrollToBottom() {
-        guard !isScrolledToBottom else { return }
+    /// Scrolls the tableView to the bottom-most row.
+    ///
+    /// This method checks if the tableView is not already scrolled to the bottom.
+    func scrollToBottomIfNeeded() {
+        let maxMessageCountForAnimation = 20
+
+        guard !isScrolledToBottom else {
+            return
+        }
 
         dataSource.loadMessages()
-        tableView.scroll(toIndex: 0)
+
+        let lastRowIndexPath = IndexPath(row: 0, section: 0)
```

Comment:
I think you should check first that there is a row at this index path.

Reply:
We check if we should continue with this method or not by checking:

If it's false then we proceed. If it's true then we return. 

```swift
var isScrolledToBottom: Bool {
        return !dataSource.hasNewerMessagesToLoad &&
        tableView.contentOffset.y + tableView.correctedContentInset.bottom <= 0
    }
``` 

Check line 61:

```swift
guard !isScrolledToBottom else {
     return
}
```

Reply:
@johnxnguyen Took care of it! Please check: https://github.com/wireapp/wire-ios/commit/f1ff30b49e7c016d6056822b8d4376f3522cfbc6